unity/Assets/Scripts/SimUtil.cs

```diff
@@ -540,7 +541,115 @@ public static class SimUtil {
             launchOnBuild ? UnityEditor.BuildOptions.AutoRunPlayer : UnityEditor.BuildOptions.None);
     }
 
-    [UnityEditor.MenuItem("Thor/Replace Generic Prefabs in All Scenes")]
+    [UnityEditor.MenuItem("AI2-THOR/RandomizeMaterials/Bin Materials")]
+    static void BinMaterials() {
+        HashSet<Material> getMaterialsInScene() {
+            HashSet<MeshRenderer> sceneMeshRenderers = new HashSet<MeshRenderer>();
+            HashSet<Material> sceneMaterials = new HashSet<Material>();
+
+            // Get all MeshRenderers under Objects and Structures groups
+            sceneMeshRenderers.UnionWith(GameObject.Find("Objects").GetComponentsInChildren<MeshRenderer>());
+            sceneMeshRenderers.UnionWith(GameObject.Find("Structure").GetComponentsInChildren<MeshRenderer>());
+
+            // Get all shared Materials from MeshRenderers
+            foreach (MeshRenderer meshRenderer in sceneMeshRenderers) {
+                sceneMaterials.UnionWith(meshRenderer.sharedMaterials);
+            }
+
+            return sceneMaterials;
+        }
+
+        Dictionary<string, HashSet<Material>> materialMetadata = new Dictionary<string, HashSet<Material>>() {
+            ["train"] = new HashSet<Material>(),
+            ["val"] = new HashSet<Material>(),
+            ["test"] = new HashSet<Material>(),
+            ["kitchen"] = new HashSet<Material>(),
+            ["livingRoom"] = new HashSet<Material>(),
+            ["bedroom"] = new HashSet<Material>(),
+            ["bathroom"] = new HashSet<Material>(),
+            ["robothor"] = new HashSet<Material>(),
+        };
+
+        // iTHOR scenes
+        foreach (int sceneType in new int[] {0, 200, 300, 400}) {
+            for (int i = 1; i <= 30; i++) {
```

Comment:
Is there a way to parameterize this `30` value in case we add more scenes in the future?

Reply:
Hmm.. probably not well. I suppose one could maybe look at the list of scenes, but suppose these scenes are in "unlisted/development only" mode (like 500 level hallways in the current set of scenes). Here, one wouldn't want to automatically include these.